test/suites/scale/deprovisioning_test.go

```diff
@@ -0,0 +1,630 @@
+				}(d)
+			}
+			wg.Wait()
+
+			// Create a separate nodeTemplate for drift so that we can change the nodeTemplate later without it affecting
+			// the other provisioners
+			driftNodeTemplate := nodeTemplate.DeepCopy()
+			driftNodeTemplate.Name = test.RandomName()
+			provisionerMap[driftValue].Spec.ProviderRef.Name = driftNodeTemplate.Name
+
+			env.MeasureDurationFor(func() {
+				By("kicking off provisioning by applying the provisioner and nodeTemplate")
+				env.ExpectCreated(driftNodeTemplate, nodeTemplate)
+				for _, p := range provisionerMap {
+					env.ExpectCreated(p)
+				}
+
+				env.EventuallyExpectCreatedMachineCount("==", expectedNodeCount)
+				env.EventuallyExpectCreatedNodeCount("==", expectedNodeCount)
+				env.EventuallyExpectInitializedNodeCount("==", expectedNodeCount)
+
+				// Wait for all pods across all deployments we have created to be in a healthy state
+				wg = sync.WaitGroup{}
+				for _, d := range deploymentMap {
+					wg.Add(1)
+					go func(dep *appsv1.Deployment) {
+						defer GinkgoRecover()
+						defer wg.Done()
+
+						env.EventuallyExpectHealthyPodCount(labels.SelectorFromSet(dep.Spec.Selector.MatchLabels), int(lo.FromPtr(dep.Spec.Replicas)))
+					}(d)
+				}
+				wg.Wait()
+			}, aws.ProvisioningEventType, multipleDeprovisionersTestGroup, defaultTestName)
+
+			env.Monitor.Reset() // Reset the monitor so that we now track the nodes starting at this point in time
+
+			By("scaling down replicas across deployments")
+			deploymentMap[consolidationValue].Spec.Replicas = lo.ToPtr[int32](int32(int(float64(replicas) * 0.2)))
+			deploymentMap[emptinessValue].Spec.Replicas = lo.ToPtr[int32](0)
+			for _, d := range deploymentMap {
+				env.ExpectUpdated(d)
+			}
+
+			env.MeasureDurationFor(func() {
+				By("enabling deprovisioning across provisioners")
+				// Create provisioners for expiration and drift so that these provisioners can do replacement
+				provisionerMap[noExpirationValue] = test.Provisioner()
+				provisionerMap[noExpirationValue].Spec = provisionerMap[expirationValue].Spec
+
+				provisionerMap[consolidationValue].Spec.Consolidation = &v1alpha5.Consolidation{Enabled: lo.ToPtr(true)}
+				provisionerMap[emptinessValue].Spec.TTLSecondsAfterEmpty = lo.ToPtr[int64](0)
+				provisionerMap[expirationValue].Spec.TTLSecondsUntilExpired = lo.ToPtr[int64](0)
+				provisionerMap[expirationValue].Spec.Limits = disableProvisioningLimits
+				for _, p := range provisionerMap {
+					env.ExpectCreatedOrUpdated(p)
+				}
+				driftNodeTemplate.Spec.AMIFamily = &v1alpha1.AMIFamilyBottlerocket
+				env.ExpectUpdated(driftNodeTemplate)
+
+				By("waiting for the nodes across all deprovisioners to get deleted")
+				type testAssertions struct {
+					deletedCount             int
+					deletedNodeCountSelector labels.Selector
+					nodeCount                int
+					nodeCountSelector        labels.Selector
+				}
+				assertionMap := map[string]testAssertions{
```

Comment:
nit: Further de-dupe you can do is to define this assertion map at the top and just re-use it for each test, parameterizing the `env.MeasureDurationFor()` with a `assertions map[string]testAssertions`

Reply:
Agreed, though (like I mentioned above), I think that this use-case right now is highly contextualized. I can see the point for generalizing this instantiation once we have n > 1 use-cases for it.